### PR TITLE
Add dynamic runtime and jdk tests

### PR DESCRIPTION
**Pull request by chyt**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,10 +5,26 @@ jdk:
     - openjdk8
     - oraclejdk8
     - oraclejdk9
-
+matrix:
+    exclude:
```
Comment (scottkurz) [inline .travis.yml]: Why are these excludes here (sorry if you already explained)?

File: boost-maven-plugin/pom.xml [hunk 2]
```diff
@@ -1,166 +1,202 @@
...
+								<goals>
+									<goal>install</goal>
+									<goal>run</goal>
+								</goals>
+							</execution>
+						</executions>
+					</plugin>
+				</plugins>
+			</build>
+		</profile>
```
Comment (scottkurz) [inline boost-maven-plugin/pom.xml]: Would be real nice if we could default from the cmd line so we didn't have to specify ` -DruntimeVersion=18.0.0.3 -Druntime=ol`.    I realize it's more work to add the logic, so could be a follow-up issue.

Comment (chyt) [reply boost-maven-plugin/pom.xml]: 18.0.0.3 is less compatible with Java 9 than 18.0.0.3, due to preparations to support Java 11 in 18.0.0.4. As a result, we don't test 18.0.0.3 on Java 9 in LMP. I think we should do the same thing here.

File: .travis.yml (hunk 1 above)
Comment (chyt) [reply .travis.yml]: 18.0.0.3 is less compatible with Java 9 than 18.0.0.2, due to changes in preparation for Java 11 support. As a result, we don't test 18.0.0.3 with Java 9 in LMP. I think we should do the same here.

File: boost-maven-plugin/pom.xml (hunk 2 above)
Comment (chyt) [reply boost-maven-plugin/pom.xml]: This is added in 2a75aaa8b400ceda0e1aec9babffc86acbebf066.

**Pull request merged**